elasticsearch-transport/lib/elasticsearch/transport/transport/http/faraday.rb

```diff
@@ -60,7 +60,12 @@ def __build_connection(host, options={}, block=nil)
           # @return [Array]
           #
           def host_unreachable_exceptions
-            [::Faraday::Error::ConnectionFailed, ::Faraday::Error::TimeoutError]
+            [
+              ::Faraday::ConnectionFailed,
+              ::Faraday::TimeoutError,
+              ::Faraday::ServerError,
```

Comment:
should be done in a backwards compatible way, `::Faraday::ServerError` is only available since 1.0 (rc atm)
current stable version of Faraday (0.15.X) does not have the `ServerError` constant, so would cause a failure - instead:
```ruby
            [
              ::Faraday::ConnectionFailed,
              ::Faraday::TimeoutError,
              ::Faraday.const_defined?(:ServerError) ? ::Faraday::ServerError : nil,
              ::Faraday::SSLError
            ].compact
```
would be also best to specify [>= qualifier to avoid future issues](https://github.com/elastic/elasticsearch-ruby/blob/1f339ac/elasticsearch-transport/elasticsearch-transport.gemspec#L30):  `s.add_dependency "faraday", ">= 0.15", "< 2"`